### DISCUSSION TITLE
Client User Context API v2

### DISCUSSION THREAD

**Discussion by camiaei** [plhery/node-twitter-api-v2#200] [Q&A] [answered]
Hi!👋 I am trying to follow this example:
https://github.com/PLhery/node-twitter-api-v2/blob/master/doc/examples.md#Streamtweetsinrealtime

I only have access to API v2, so I replaced method v1 with [v2](https://github.com/PLhery/node-twitter-api-v2/blob/master/doc/v2.md#Replytoatweet), but I get this error:

 "Authenticating with OAuth 2.0 Application-Only is forbidden for this endpoint.  Supported authentication types are [OAuth 1.0a User Context, OAuth 2.0 User Context]."

I have tried instantiating the client in different ways, but it doesn't work😓
I know I am doing something wrong, I would appreciate if someone could help me.

https://gist.github.com/camiaei/41845bb517da261dc15a51218957e4a2

**Comment by alkihis** [comment, marked as answer]
Hi :wave: !
As specified in [twitter documentation](https://developer.twitter.com/en/docs/twitter-api/tweets/filtered-stream/api-reference/get-tweets-search-stream), stream endpoint only supports v2 app-only authentication.

On the other side, reply to a tweet requires a client with user-context authentication... :smile: 

So, you need to have two different clients:
One client with a bearer token (your first tentative) to use tweet stream, and the other client (your second one), with user-context (in your case, OAuth 1.0a access tokens) to reply to tweet !

**Reply by camiaei** [reply]
> and the other client (your second one), with user-context (in your case, OAuth 1.0a access tokens) to reply to tweet !

Can I do this part with OAuth 2.0 User Context?
(I only have access to the v2 API endpoints 😓)

**Reply by alkihis** [reply]
Of course! You will need to generate a Bearer token with your client ID/client secret.

**Reply by camiaei** [reply]
![oauth_1a](https://user-images.githubusercontent.com/79164595/155808556-65fbc5f3-49bb-43d5-bfcf-44813d47fd7f.png)

It worked. I activated that☝ and instantiated the two clients. 

Q: If I'm using `client.v2.reply()`, that means I'm hitting the new endpoints right? (I'm very confused between the API version and the OAuth version😵)

Also, [I found this](https://github.com/alkihis/twitter-api-v2-user-oauth-flow-example), which I had also been trying to do❤
